tools/cmake/FindBpfProg.cmake

```diff
@@ -0,0 +1,182 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#[=======================================================================[.rst:
+FindBpfProg
+--------
+
+Find BpfProg
+
+This module finds if all the dependencies for eBPF Compile-Once-Run-Everywhere
+programs are available and where all the components are located.
+
+The caller may set the following variables to disable automatic
+search/processing for the associated component:
+
+  ``BPFTOOL_EXE``
+    Path to ``bpftool`` binary
+
+  ``CLANG_EXE``
+    Path to ``clang`` binary
+
+  ``LIBBPF_INCLUDE_DIRS``
+    Path to ``libbpf`` development headers
+
+  ``LIBBPF_LIBRARIES``
+    Path to `libbpf` library
+
+  ``GENERATED_VMLINUX``
+    Path to ``bpftool`` generated ``vmlinux.h`` file. If unset, this module will
+    attempt to automatically generate a copy.
+
+This module sets the following result variables:
+
+::
+
+  BpfProg_FOUND             = TRUE if all components are found
+
+
+This module also provides the ``bpf_prog()`` macro. This macro generates a
+cmake interface library for the BPF program's generated skeleton as well
+as the associated dependencies.
+
+.. code-block:: cmake
+
+  bpf_prog(<name> <source>)
+
+Given an abstract ``<name>`` for a BPF program and the associated ``<source>``
+file, generates an interface library target, ``<name>_prog``, that may be
```

Comment:
I'd never guess that _prog means BPF object file. How about `<name>_bpf`? Or wait, this is skeleton stuff, no? If skeleton, then `<name>_skel`?

Reply:
_skel sounds better